lua/fidget/progress/handle.lua

```diff
@@ -0,0 +1,145 @@
+local notification = require("fidget.notification")
+
+local NEXT_ID = 0
+local PREFIX = "fidget-"
+
+local function next_token()
+  NEXT_ID = NEXT_ID + 1
+  return PREFIX .. NEXT_ID
+end
+
+--- @class ProgressHandle: ProgressMessage  A handle for a progress message, reactive to changes
+--- @field cancel fun(self: ProgressHandle) Cancel the task
+--- @field finish fun(self: ProgressHandle) Mark the task as complete
+--- @field report fun(self: ProgressHandle, props: { [string]: any } | ProgressMessage) Update one or more properties of the progress message
+
+local M = {}
+
+--- Create a new progress message, and return a handle to it for updating.
+--- The handle is a reactive object, so you can update its properties and the
+--- message will be updated accordingly. You can also use the `report` method to
+--- update multiple properties at once.
+---
+--- Example:
+---
+--- ```lua
```

Comment:
Nit: this needs to be `--->lua`, i.e., use vimhelp formatting, since lemmy-help doesn't do any translation from Markdown.